contracts/superfluid/CFALib.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {ISuperToken} from "@superfluid-finance/ethereum-contracts/contracts/interfaces/superfluid/ISuperfluid.sol";
+import {IConstantFlowAgreementV1} from "@superfluid-finance/ethereum-contracts/contracts/interfaces/agreements/IConstantFlowAgreementV1.sol";
+import {CFAv1Library} from "@superfluid-finance/ethereum-contracts/contracts/apps/CFAv1Library.sol";
+
+library CFALib {
+    using CFAv1Library for CFAv1Library.InitData;
+
+    function _decreaseFlow(
+        CFAv1Library.InitData storage cfaV1,
+        IConstantFlowAgreementV1 _cfa,
+        ISuperToken token,
+        address to,
+        int96 by
+    ) internal {
+        (, int96 curRate, , ) = _cfa.getFlow(token, address(this), to);
+        int96 newRate = curRate - by;
+        require(newRate >= 0, "new rate would be negative");
+
+        if (newRate == 0) {
+            cfaV1.deleteFlow(address(this), to, token);
+        } else {
+            cfaV1.updateFlow(to, token, newRate);
+        }
+    }
+
+    function _increaseFlow(
+        CFAv1Library.InitData storage cfaV1,
+        IConstantFlowAgreementV1 _cfa,
+        ISuperToken token,
+        address to,
+        int96 by
+    ) internal {
+        (, int96 curRate, , ) = _cfa.getFlow(token, address(this), to);
+        int96 newRate = curRate + by;
+        require(newRate >= 0, "overflow");
```

Comment:
Would solidity automatically report error in the previous stmt if there is an overview? Thus, making this statement useless?

Reply:
I think a check newRate > curRate will work better here.

Reply:
You are right, solidity above 0.8 can handle math overflow automatically. Remove this line.